backend/gsr_booking/api_wrapper.py

```diff
@@ -532,38 +311,175 @@ def get_affiliation(self, email):
         else:
             return "Other"
 
-    def cancel_room(self, user, booking_id):
-        """Cancels room"""
-        gsr_booking = get_object_or_404(GSRBooking, booking_id=booking_id)
-        if gsr_booking:
-            if user != gsr_booking.user and user != gsr_booking.reservation.creator:
+            
+class BookingHandler:
+    def __init__(self, WBW=None, LBW=None):
+        self.WBW = WBW or WhartonBookingWrapper()
+        self.LBW = LBW or LibCalBookingWrapper()
+
+
+    def format_members(self, members):
+        return [
+            (User(**{ k[6:]: v for k, v in member.items() if k.startswith('user__') }), # temp user object
+            member['left'])
+            for member in members
+        ]
+
+    def get_wharton_members(self, group, gsr_id):
+        now = timezone.localtime()
+        ninty_min = datetime.timedelta(minutes=90)
+        zero_min = datetime.timedelta(minutes=0)
+        ret = GroupMembership.objects.filter(group=group, is_wharton=True).values('user').annotate(left=ninty_min-Coalesce(Sum(
```

Comment:
if we are removing this `is_wharton` cache, how would we change this code? perhaps we just keep the `is_wharton` field for Labs people?

Reply:
we are still using is_wharton for groups, which we would manage ourselves since we only have the 1 Penn Labs group. 
Otherwise our queries would be really expensive.